packages/lib/BaseApplication.ts

```diff
@@ -927,6 +929,36 @@ export default class BaseApplication {
 
 		await MigrationService.instance().run();
 
+		this.startUpRotateLogFilesRoutine(profileDir);
+
 		return argv;
 	}
+
+	private startUpRotateLogFilesRoutine(profileDir: string) {
+		const rl: RotationalLogs = new RotatingLogs(profileDir);
+		this.cleanActiveLogWrapper(rl);
+		this.deleteNonActiveLogFilesWrapper(rl);
+	}
+
+	private cleanActiveLogWrapper(rl: RotationalLogs) {
+		const sixtySeconds: number = 60 * 1000;
+		shim.setTimeout(async () => {
+			try {
+				await rl.cleanActiveLogFile();
+			} catch (e) {
+				appLogger.error('Cannot clean the active log file.', e);
+			}
+		}, sixtySeconds);
+	}
+
+	private deleteNonActiveLogFilesWrapper(rl: RotationalLogs) {
+		const twentyFourHours: number = 60 * 60 * 24 * 1000;
+		shim.setInterval(async () => {
+			try {
+				await rl.deleNonActiveLogFiles();
+			} catch (e) {
+				appLogger.error('Cannot delete the non active log file(s).', e);
+			}
+		}, twentyFourHours);
+	}
```

Comment:
I'm so confused with the complexity of this code Hubert. Like... you create a new instance of `RotatingLogs`, but it doesn't return a new instance of `RotatingLogs` but something else called `RotationalLogs`?

Also I feel that's a lot of lines for not much. Here's what we need:

```typescript
this.rotatingLogs = new RotatingLogs(path);

const processLogs = async () => {
	try {
		await this.rotatingLogs.cleanActiveLogFile();
		await this.rotatingLogs.deleNonActiveLogFiles();
	} catch (error) {
		appLogger.error(error);
	}
}

// Then you can use the same pattern I've shown you earlier:

shim.setTimeout(() => { void processLogs(); }, 5000);
shim.setInterval(() => { void processLogs(); }, 12 * 60 * 60 * 1000);
```